sqldatabaseproductserver/main.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	_ "github.com/mattn/go-sqlite3"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+type Product struct {
+	ID    int    `json:"id"`
+	Name  string `json:"name"`
+	Price int    `json:"price"`
+}
+
+var products []Product
+
+func main() {
+	db, err := sql.Open("sqlite3", "products.db")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	err = db.Ping()
+	if err != nil {
+		fmt.Println("error pinging database")
+		return
+	}
+	r := mux.NewRouter()
+	r.HandleFunc("/products", GetProducts).Methods("GET")
+	r.HandleFunc("/products/{id}", GetProduct).Methods("GET")
+	r.HandleFunc("/products", AddProduct).Methods("POST")
+	r.HandleFunc("/products/{id}", UpdateProduct).Methods("PUT")
+	r.HandleFunc("/products/{id}", DeleteProduct).Methods("DELETE")
+	fmt.Println("Server is running")
+	log.Fatal(http.ListenAndServe("localhost:8080", r))
+
+}
+
+func GetProducts(w http.ResponseWriter, r *http.Request) {
+	db, err := sql.Open("sqlite3", "products.db")
```

Comment:
Use a global for `db`. No need to open a new connection to the database on every request.